File: python/multi_thread.py
import time
from threading import Thread
    
def convert_multi_threaded():
    def convert_single_threaded():
        start_time = time.time()

        #! Lógica de conversão de documentos aqui

        end_time = time.time()
        execution_time = end_time - start_time
        print(f"Tempo de execução (single-threaded): {execution_time} segundos")
    start_time = time.time()

    # Criar threads para simular conversão simultânea
    threads = [Thread(target=convert_single_threaded) for _ in range(5)]  # Exemplo com 5 threads
    for thread in threads:
        thread.start()
    for thread in threads:
        thread.join()

    end_time = time.time()
    execution_time = end_time - start_time
    print(f"Tempo de execução (multi-threaded): {execution_time} segundos")

if __name__ == "__main__":
    convert_multi_threaded()

File: java/multi_thread.java
import java.util.concurrent.CountDownLatch;

public class multi_thread {

    private static void convertMultiThreaded() {
        long startTime = System.currentTimeMillis();

        // Criar wait group para aguardar a conclusão de todas as threads
        int numberOfThreads = 5;  // Exemplo com 5 threads
        Thread[] threads = new Thread[numberOfThreads];
        CountDownLatch latch = new CountDownLatch(numberOfThreads);

        for (int i = 0; i < numberOfThreads; i++) {
            threads[i] = new Thread(() -> {
                //! Chamar a função de conversão
                latch.countDown();
            });
            threads[i].start();
        }

        // Aguardar a conclusão de todas as threads
        try {
            latch.await();
        } catch (InterruptedException e) {
            e.printStackTrace();
        }

        long endTime = System.currentTimeMillis();
        double executionTime = (endTime - startTime) / 1000.0; // em segundos
        System.out.println("Tempo de execução (multi-threaded): " + executionTime + " segundos");
    }

    public static void main(String[] args) {
        convertMultiThreaded();
    }
}
